CorsixTH/Lua/world.lua

```diff
@@ -1095,7 +1096,7 @@ function World:onEndDay()
   --check if it's time for a VIP visit
   if self.game_date:isSameDay(self.next_vip_date) then
     if #self.rooms > 0 and self.ui.hospital:hasStaffedDesk() then
-      self.hospitals[1]:createVip()
+      local_hospital:createVip()
```

Comment:
In my view, an object should not make decisions for another object without that explicit as design decision.
Here, world thus shouldn't decide for hospital.

All objects are grown-ups, they can decide things for themselves. If an object needs a decision, ask the other object about it. Don't make a decision without consulting the object that is affected.

Basically, apply real-life courtesy.


In code, that means you write a function in the hospital class that decides it is ready for a VIP visit, and call that.
[If you move the entire scheduling into hospital, the guideline also applies, but since the code is completely in the affected object then, it can simply compute the result directly (for short code at least).]

Reply:
I think this is going outside the scope of the PR

Reply:
That's valid way of looking at it